sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -136,6 +136,7 @@ class Analyzer(
       ResolveGroupingAnalytics ::
       ResolvePivot ::
       ResolveOrdinalInOrderByAndGroupBy ::
+      ResolveAggAliasInGroupBy ::
```

Comment:
is it safer to put it in an individual batch after the `resolution` batch? Ideally we should only run this rule if we make sure there is no other way to resolve the grouping expressions exception this rule. cc @gatorsmile